packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1732,29 +1732,70 @@ class TransformLayer extends OffsetLayer {
 class OpacityLayer extends OffsetLayer {
   /// Creates an opacity layer.
   ///
-  /// The [alpha] property must be non-null before the compositing phase of
-  /// the pipeline.
+  /// Either the [alpha] or [opacity] property must be non-null before the compositing phase of
+  /// the pipeline. Providing an [alpha] is deprecated and [opacity] should be used
+  /// instead.
   OpacityLayer({
+    @Deprecated(
+      'Prefer using opacity instead of alpha. '
+      'This feature was deprecated after v2.13.0-0.0.pre.'
+    )
     int? alpha,
+    double? opacity,
     super.offset,
-  }) : _alpha = alpha;
+  }) : assert(
+      (alpha == null && opacity == null) ||
+      (alpha == null && opacity != null) ||
+      (alpha != null && opacity == null),
+      'Only one of alpha and opacity should be provided.'
+      ),
+      _opacity = opacity ?? (alpha != null ? _alphaToOpacity(alpha) : null);
+
+  static double _alphaToOpacity(int value) {
+    return value / 255.0;
+  }
+
+  static int _alphaFromOpacity(double value) {
+    return (value * 255).round();
+  }
 
   /// The amount to multiply into the alpha channel.
   ///
   /// The opacity is expressed as an integer from 0 to 255, where 0 is fully
   /// transparent and 255 is fully opaque.
   ///
+  /// Providing an [alpha] is deprecated and [opacity] should be used
+  /// instead.
+  ///
   /// The scene must be explicitly recomposited after this property is changed
   /// (as described at [Layer]).
-  int? get alpha => _alpha;
-  int? _alpha;
+  @Deprecated(
+    'Prefer using opacity instead of alpha. '
+    'This feature was deprecated after v2.13.0-0.0.pre.'
+  )
+  int? get alpha => _opacity == null ? null : _alphaFromOpacity(_opacity!);
   set alpha(int? value) {
     assert(value != null);
-    if (value != _alpha) {
-      if (value == 255 || _alpha == 255) {
+    opacity = _alphaToOpacity(value!);
+  }
+
+  /// The opacity to composite with child layers.
+  ///
+  /// The opacity is expressed as a double from 0.0 to 1.0, where 0.0 is fully
+  /// transparent and 1.0 is fully opaque.
+  ///
+  /// The scene must be explicitly recomposited after this property is changed
+  /// (as described at [Layer]).
+  double? get opacity => _opacity;
+  double? _opacity;
+  set opacity(double? value) {
+    assert(value != null);
```

Comment:
Should this just use a non-nullable type instead?

Reply:
I looked into changing this by using `late`, but then that doesn't allow whether the field has been set to be observed so we can't compare the new value to the old in the setter safely. We could default to 0.0 or 1.0 but that would make current misuse of the API that throws an exception instead silently do ... something else.

There is likely a way we can re-arrange this API to work better but I think it would be best as part of a separate effort, so that we can cover all of the layer types

Reply:
I meant as an argument type. The internal fields could be nullable if that makes things easier, but having an "assert(argument != null)" as the first line in a method indicates that arg should probably have been expressed as non-nullable...